### PR TITLE
Implement functions to clone rubric into API

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add backend functions for replacing ORA rubric editor data with data from another rubric.

JIRA: [EDUCATOR-5754](https://openedx.atlassian.net/browse/EDUCATOR-5754)

**What changed?**

- Adds a method to `oa_server.js` for getting rubric data from ORA backend
- Adds methods in `oa_edit_rubric.js` to clear and replace rubric data with returned data from backend
- Adds a method in `oa_edit_listeners.js` for clearing learner training examples when a rubric is replaced
- **TEMP:** expose a dev API to the window for test invocation (re-enable by applying [this diff](https://github.com/edx/edx-ora2/files/6704632/rubric_dev_hook.diff.txt))

**What didn't change?** 
I did not add tests for the following, since unit tests either do not exist or do not cover appropriate areas in these files:
  - `oa_edit_container_item.js`
  - `oa_edit.js`
  - `oa_edit_fields.js` (there is a TODO already marked here)

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

Testing is done through an exposed dev API in the browser.

1. Setup an ORA with a non-default rubric. Play with different numbers of criteria/options.
2. Get the ORA location ID by hooking/logging `other_openassessments_in_course` from Studio context
(note: this is different for published and non-published ORAs and can change when an ORA is updated)
3. Create a new ORA in Studio and click the "Edit" button to open the editor window
4. Open the browser developer console and enter the following code `window.__dev__.cloneRubric(<rubric-location>)`
5. Verify that rubric is replaced with new data
6. Verify that the learner training examples are cleared

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/static/js/src/studio/oa_edit_fields.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ export const Fields = {
     if (value !== undefined) { sel.prop('checked', value); }
     return sel.prop('checked');
   },
+
+  selectField: (sel, value) => {
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/studio/oa_edit_fields.js:16]: Add a docstring here

File: openassessment/xblock/static/js/src/oa_server.js [hunk 2]
```diff
@@ -705,6 +705,30 @@ export class Server {
...
+  /**
+   * Clone a rubric into ORA from an existing rubric
+   * @param {xblock-id} rubricLocation, xblock locator for ORA to clone rubric from
+   * @returns {promise} a JQuery Promise which will resolve with rubric data
+   */
+  cloneRubric(rubricLocation) {
+    const url = this.url('get_rubric');
+    const payload = { target_rubric_block_id: String(rubricLocation) };
+
+    return $.Deferred((defer) => {
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/oa_server.js:718]: Investigate replacing with `axios` or `fetch`

File: openassessment/xblock/static/js/src/studio/oa_edit_rubric.js [hunk 3]
```diff
@@ -33,6 +35,64 @@ export class EditRubricView {
...
+
+    // For dev use, expose to browser
+    window.__dev__ = this;
+  }
+
+  /**
+   * Overwrite ORA rubric from existing rubric at location {rubricLocation}
+   * @param {xblock-id} rubricLocation
+   */
+  cloneRubric(rubricLocation) {
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/studio/oa_edit_rubric.js:47]: Split into smaller pieces :)

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js [hunk 4]
```diff
@@ -277,6 +277,15 @@ export class StudentTrainingListener {
       .text(data.label);
   }
 
+  /**
+   * Actions to take when a rubric is cloned/replaced. Currently this is:
+   *  - Clear training examples
+   */
+  rubricReplaced() {
+    const trainingExamples = $('#openassessment_training_example_list .openassessment_training_example');
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/studio/oa_edit_listeners.js:285]: Make sure this is just searching within the target rubric.

File: openassessment/xblock/static/js/src/oa_server.js (hunk 2 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/src/oa_server.js:718]: Ummm... I'm gonna call this out of scope since everything else here uses this format.

File: openassessment/xblock/static/js/spec/studio/oa_edit.js [hunk 5]
```diff
@@ -47,7 +47,7 @@ describe("OpenAssessment.StudioView", function() {
     var EXPECTED_SERVER_DATA = {
         title: "The most important of all questions.",
         prompts: [{"description": "How much do you like waffles?"}, {description : 'How much do you like waffles 2?'}],
-        feedbackPrompt: "",
```
Comment (jansenk) [inline openassessment/xblock/static/js/spec/studio/oa_edit.js:50]: nothing for feedback_default_text?

File: openassessment/xblock/static/js/fixtures/templates.json [hunk 6]
```diff
@@ -488,6 +488,8 @@
                     "points_possible": 2
                 }
             ],
+            "feedbackprompt": "Feedback default prompt",
```
Comment (jansenk) [inline openassessment/xblock/static/js/fixtures/templates.json:491]: I see this as `feedback_prompt` and `feedbackPrompt` elsewhere in the PR. why is it `feedbackprompt` here?

**Review by jansenk — APPROVED** [review]
Visual review looks good! Running through some local tests.

File: openassessment/xblock/static/js/fixtures/templates.json (hunk 6 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/fixtures/templates.json:491]: There appears to be a naming inconsistency between the way data is stored (Rubric JSON and XML) and the variable in context used to populate the template, which uses `feedbackprompt`: https://github.com/edx/edx-ora2/blob/a82c2cdc6804d36642279959fe2d27a94882fc1b/openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html#L38

Comment (jansenk) [reply openassessment/xblock/static/js/fixtures/templates.json:491]: wonderful

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js (hunk 4 above)
Comment (leangseu-edx) [reply openassessment/xblock/static/js/src/studio/oa_edit_listeners.js:285]: To ensure it is select within, you should pass `this.elements` as the second argument for this selector.

File: openassessment/xblock/static/js/src/oa_server.js (hunk 2 above)
Comment (muselesscreator) [reply openassessment/xblock/static/js/src/oa_server.js:718]: new ticket for replacing network comms may make sense, or at least cataloguing them

File: openassessment/xblock/static/js/fixtures/templates.json (hunk 6 above)
Comment (muselesscreator) [reply openassessment/xblock/static/js/fixtures/templates.json:491]: js expects camelCase while python expects snake_case, which results in this kind of mismatch rather often unfortunately.

File: openassessment/xblock/static/js/spec/oa_server.js [hunk 7]
```diff
@@ -56,6 +56,43 @@ describe("OpenAssessment.Server", function() {
         '</criterion>'+
     '</rubric>';
 
+    var RUBRIC_JSON = {
+        "criteria": [
+            {
+                "label": "𝓒𝓸𝓷𝓬𝓲𝓼𝓮",
```
Comment (muselesscreator) [inline openassessment/xblock/static/js/spec/oa_server.js:62]: dude... wat?

**Review by muselesscreator — APPROVED** [review]
LGTM

File: openassessment/xblock/static/js/spec/studio/oa_edit.js (hunk 5 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/spec/studio/oa_edit.js:50]: Whoops, the test wasn't originally there for feedback_default_text. Just added.

File: openassessment/xblock/static/js/spec/oa_server.js (hunk 7 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/spec/oa_server.js:62]: Lol, this was an existing example, so I copied it. Not sure what cursed back room they got those characters from but I'm about it.
